Source/AppAuthCore/OIDAuthState.m

```diff
@@ -442,7 +442,54 @@ - (OIDTokenRequest *)tokenRefreshRequestWithAdditionalParameters:
                       scope:nil
                refreshToken:_refreshToken
                codeVerifier:nil
-       additionalParameters:additionalParameters];
+       additionalParameters:additionalParameters
+          additionalHeaders:nil];
+}
+
+- (OIDTokenRequest *)tokenRefreshRequestWithAdditionalParameters:
+    (NSDictionary<NSString *, NSString *> *)additionalParameters
+                                               additionalHeaders:
+    (NSDictionary<NSString *,NSString *> *)additionalHeaders {
+
+  // TODO: Add unit test to confirm exception is thrown when expected
```

Comment:
Is this a test that you intend to write in the future? If so, I'd prefer to see this `TODO` annotated with an issue assigned to you. Or, perhaps you intended to write this test and remove the `TODO` in this PR?

Reply:
I'm not sure what the purpose or thinking behind the `TODO` was. I see two options:

1. Remove the `TODO`s you added since you don't have a plan for them.
2. Add a test that creates an instance of `OIDAuthState` using [the designated initializer](https://github.com/openid/AppAuth-iOS/blob/master/Source/AppAuthCore/OIDAuthState.h#L164). You can pass nil to the initializer's [tokenResponse parameter](https://github.com/openid/AppAuth-iOS/blob/master/Source/AppAuthCore/OIDAuthState.m#L215) which will prevent `-[OIDAuthState updateWithTokenResponse:error:]` from being called, and will therefore [not set the `_refreshToken`](https://github.com/openid/AppAuth-iOS/blob/master/Source/AppAuthCore/OIDAuthState.m#L407). From there, once you have your instance of `OIDAuthState` with a nil `_refreshToken`, you should be able to write a test the verifies an exception is thrown when you call `-[OIDAuthState tokenRefreshRequestWithAdditionalParameters:additionalHeaders:]`.

If this works, would you mind adding similar tests for the existing `TODO`s and removing the comments?

Assuming the second option is as straightforward as I describe, then I think this is the way to go.

Reply:
This actually helped. Thanks a lot.

I've removed all TODOs in `OIDAuthState.m` and added the tests accordingly.